### PR TITLE
Enable caching for ANTLR task

### PR DESCRIPTION
**Pull request by lptr**

### CODE REVIEW

File: subprojects/docs/src/docs/release/notes.md [hunk 1]
```diff
@@ -48,6 +48,10 @@ In this release, Gradle's incremental C/C++ compilation is now able to understan
 
 When registering task properties via the runtime API, property names are not required to be Java identifiers anymore, and can be any non-empty string.
 
+### ANTLR task is now cacheable by default
+
+When generating grammar sources with ANTLR, now the task's outputs are stored and retreived from the build cache.
```
Comment (big-guy) [inline subprojects/docs/src/docs/release/notes.md]: misspelling, `retrieved`

Comment (big-guy) [reply subprojects/docs/src/docs/release/notes.md]: Should we also add this task to the list of cacheable tasks in the build cache chapter?

File: subprojects/antlr/src/integTest/groovy/org/gradle/api/plugins/antlr/AntlrRelocationIntegrationTest.groovy [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.plugins.antlr
+
+import org.gradle.integtests.fixtures.AbstractTaskRelocationIntegrationTest
+
+class AntlrRelocationIntegrationTest extends AbstractTaskRelocationIntegrationTest {
```
Comment (big-guy) [inline subprojects/antlr/src/integTest/groovy/org/gradle/api/plugins/antlr/AntlrRelocationIntegrationTest.groovy]: Does this need to be tested against each of the Antlr versions Gradle supports (2,3,4)?

File: subprojects/docs/src/docs/release/notes.md (hunk 1 above)
Comment (lptr) [reply subprojects/docs/src/docs/release/notes.md]: Good point, added, fixed.

File: subprojects/antlr/src/integTest/groovy/org/gradle/api/plugins/antlr/AntlrRelocationIntegrationTest.groovy (hunk 2 above)
Comment (lptr) [reply subprojects/antlr/src/integTest/groovy/org/gradle/api/plugins/antlr/AntlrRelocationIntegrationTest.groovy]: Added tests for each ANTLR versions.

**Comment by lptr** [comment]
Thanks. I've added coverage for each ANTLR version. The timestamp is irrelevant in this case: it doesn't make caching better or worse for the ANTLR task. We _could_ improve things downstream by removing the timestamp, but that's a separate concern.

**Comment by lptr** [comment]
Feedback build: https://builds.gradle.org/viewQueued.html?itemId=9862568

**Comment by lptr** [comment]
Feature build: https://builds.gradle.org/viewQueued.html?itemId=9863268

**Comment by lptr** [comment]
Feedback build: https://builds.gradle.org/viewQueued.html?itemId=9865582

**Comment by lptr** [comment]
Feedback build: https://builds.gradle.org/viewQueued.html?itemId=9866030

**Comment by lptr** [comment]
I re-ran the build that timed out and it's fine. Merging.

**Pull request merged**